### PR TITLE
Add `mc-sgx-sync::sys::locks::Condvar`

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Add the backend end `Condvar` implementation for use in SGX enclaves

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #14** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/14" target="_blank">Graphite</a> 
    * **PR #13** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/13" target="_blank">Graphite</a> 
      * **PR #15** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/15" target="_blank">Graphite</a> 
        * **PR #22** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/22" target="_blank">Graphite</a> 
          * **PR #24** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/24" target="_blank">Graphite</a> 
            * **PR #30** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/30" target="_blank">Graphite</a> 
              * **PR #31** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/31" target="_blank">Graphite</a> 
                * **PR #32** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/32" target="_blank">Graphite</a> 
                  * **PR #33** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/33" target="_blank">Graphite</a> 
                    * **PR #34** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/34" target="_blank">Graphite</a> 
                      * **PR #35** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/35" target="_blank">Graphite</a> 
                        * **PR #36** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/36" target="_blank">Graphite</a> 
                          * **PR #39** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/39" target="_blank">Graphite</a> 
                            * **PR #40** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/40" target="_blank">Graphite</a> 
                              * **PR #41** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/41" target="_blank">Graphite</a> 
                                * **PR #42** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/42" target="_blank">Graphite</a>  👈
        * **PR #21** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/21" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/42?utm_source=stack-comment).

**Review by awygle — APPROVED** [review]
LGTM.

File: sync/src/sys/locks/condvar.rs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+use mc_sgx_tstdc::Condvar as SgxCondvar;
+
+/// The condition variable backend to use with the common Rust std lib Condvar
+/// interface
+pub(crate) struct Condvar {
+    inner: SgxCondvar,
+}
+
+unsafe impl Send for Condvar {}
+unsafe impl Sync for Condvar {}
```
Comment (jcape) [inline sync/src/sys/locks/condvar.rs:15]: Advantages/disadvantages of implementing these on `mc_sgx_tstdc::Condvar`?

**Review by jcape — APPROVED** [review]
LGTM, question about the manual Send/Sync impl.

File: sync/src/sys/locks/condvar.rs (hunk 1 above)
Comment (nick-mobilecoin) [reply sync/src/sys/locks/condvar.rs:15]: blarg, I forgot they are automatically derived, https://doc.rust-lang.org/nomicon/send-and-sync.html

Should probably do the same for Mutex and RwLock, I'll create a follow on PR to remove the unnecessary traits at this level.
Added as a task to #29

**Comment by nick-mobilecoin** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Stack job ID: uCoNH97G95Be2subCEb9.
See details on [graphite.dev](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/42)

**Pull request merged**